en/views/helpers/form.rst

```diff
@@ -91,18 +107,62 @@ Output:
 
 .. note::
 
-    Since this is an edit form, a hidden input field is generated to
+    Since this is an *edit* form, a hidden ``input`` field is generated to
     override the default HTTP method.
 
+Options for Form
+----------------
+
 The ``$options`` array is where most of the form configuration
 happens. This special array can contain a number of different
 key-value pairs that affect the way the form tag is generated.
+Valid values:
+
+* ``'type'`` - Allows you to choose the type of form to create. If no type is
+  provided then it will be autodetected based on the form context.
+  Valid values:
+
+  #. ``'get'`` - Will set the form method to HTTP GET.
+  #. ``'file'`` - Will set the form method to POST and the ``'enctype'`` to
+     "multipart/form-data".
+  #. ``'post'`` - Will set the method to POST.
+  #. ``'put', 'delete', 'patch'`` - Will override the HTTP method with PUT,
+     DELETE or PATCH respectively, when the form is submitted.
+
+* ``'method'`` - Valid values are the same as above. Allows you to explicitly
+  override the form's method.
+
+* ``'url'`` - Specify the URL the form will submit to. Can be a string or a URL
+  array.
+
+* ``'encoding'`` - Sets the ``accept-charset`` encoding for the form. Defaults
+  to ``Configure::read('App.encoding')``.
+
+* ``'enctype'`` - Allows you to set the form encoding explicitly.
+
+* ``'templates'`` - The templates you want to use for this form. Any templates
+  provided will be merged on top of the already loaded templates. Can be either
+  a filename from ``/config`` or an array of templates to use.
+
+* ``'context'`` - Additional options for the form context class. (For example
+  the ``EntityContext`` accepts a ``'table'`` option that allows you to set the
+  specific Table class the form should be based on.)
+
+* ``'idPrefix'`` - Prefix for generated ID attributes.
 
+* ``'templateVars'`` - Allows you to provide template variables for the
+  ``formStart`` template.
+
+.. tip::
+
+    Besides the above options you can provide, in the ``$options`` argument,
+    any valid HTML attributes that you want to pass to the created ``form``
+    element.
 
 Changing the HTTP Method for a Form
 -----------------------------------
 
-By using the ``type`` option you can change the HTTP method a form will use::
+By using the ``'type'`` option you can change the HTTP method a form will use::
```

Comment:
We don't generally include single quotes around key name as the inline code indicates that it is a key name.